src/contexts/task-context.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {
+  Context,
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+import { TaskService } from '../services';
+import { AccessToken, Task } from '../types';
+
+import { useAuthContext } from './auth-context';
+
+interface TaskContextInterface {
+  addTask: (description: string, title: string) => Promise<void>;
+  deleteTask: (task: Task) => Promise<void>;
+  getTasks: () => Promise<void>;
+  isAddTaskLoading: boolean;
+  isDeleteTaskLoading: boolean;
+  isTasksLoading: boolean;
+  isUpdateTaskLoading: boolean;
+  tasks: Task[];
+  updateTask: (task: Task) => Promise<void>;
+}
+
+export const TaskContext = createContext<TaskContextInterface | undefined>(undefined);
+
+export const useTaskContext = (): TaskContextInterface =>
+  useContext(TaskContext as Context<TaskContextInterface>);
+
+export const TaskContextProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const taskService = useMemo(() => new TaskService(), []);
+
+  const [tasks, setTasks] = useState<Task[]>([]);
+  const [isTasksLoading, setIsTasksLoading] = useState(false);
+  const [isAddTaskLoading, setIsAddTaskLoading] = useState(false);
+  const [isUpdateTaskLoading, setIsUpdateTaskLoading] = useState(false);
+  const [isDeleteTaskLoading, setIsDeleteTaskLoading] = useState(false);
+
+  const { getAccessTokenFromStorage } = useAuthContext();
```

Comment:
Though this is not related to this change, let's rename `getAccessTokenFromStorage` to `getAccessToken` instead. The caller does not needs to know from _where_ (ex - local storage) the token is being fetched from from. Same can be done for related functionalities:

- setAccessTokenToStorage -> setAccessToken
- clearAccessTokenFromStorage -> deleteAccessToken

Reply:
Renamed them.